File: kmp/composeApp/src/commonTest/kotlin/com/tanyudii/tmuxweb/domain/TerminalSearchTest.kt
package com.tanyudii.tmuxweb.domain

import kotlin.test.Test
import kotlin.test.assertFalse
import kotlin.test.assertTrue

class TerminalSearchTest {
    @Test
    fun `recognizes Ctrl+F on keydown`() {
        val result = isFindShortcut(type = "keydown", ctrlKey = true, metaKey = false, key = "f")

        assertTrue(result)
    }

    @Test
    fun `recognizes Cmd+F on keydown`() {
        val result = isFindShortcut(type = "keydown", ctrlKey = false, metaKey = true, key = "f")

        assertTrue(result)
    }

    @Test
    fun `is case-insensitive on the key value`() {
        val result = isFindShortcut(type = "keydown", ctrlKey = true, metaKey = false, key = "F")

        assertTrue(result)
    }

    @Test
    fun `ignores keyup so search doesn't retrigger per press`() {
        val result = isFindShortcut(type = "keyup", ctrlKey = true, metaKey = false, key = "f")

        assertFalse(result)
    }

    @Test
    fun `ignores plain f with no modifier so typing still reaches the shell`() {
        val result = isFindShortcut(type = "keydown", ctrlKey = false, metaKey = false, key = "f")

        assertFalse(result)
    }

    @Test
    fun `ignores unrelated Ctrl shortcuts`() {
        val result = isFindShortcut(type = "keydown", ctrlKey = true, metaKey = false, key = "c")

        assertFalse(result)
    }
}

File: web/src/domain/terminalSearch.test.ts
import { describe, expect, test } from "vitest";
import { isFindShortcut } from "./terminalSearch";

// Ports kmp/.../domain/TerminalSearchTest.kt 1:1.
describe("isFindShortcut", () => {
  test("recognizes Ctrl+F on keydown", () => {
    expect(isFindShortcut({ type: "keydown", ctrlKey: true, metaKey: false, key: "f" })).toBe(true);
  });

  test("recognizes Cmd+F on keydown", () => {
    expect(isFindShortcut({ type: "keydown", ctrlKey: false, metaKey: true, key: "f" })).toBe(true);
  });

  test("is case-insensitive on the key value", () => {
    expect(isFindShortcut({ type: "keydown", ctrlKey: true, metaKey: false, key: "F" })).toBe(true);
  });

  test("ignores keyup so search doesn't retrigger per press", () => {
    expect(isFindShortcut({ type: "keyup", ctrlKey: true, metaKey: false, key: "f" })).toBe(false);
  });

  test("ignores plain f with no modifier so typing still reaches the shell", () => {
    expect(isFindShortcut({ type: "keydown", ctrlKey: false, metaKey: false, key: "f" })).toBe(false);
  });

  test("ignores unrelated Ctrl shortcuts", () => {
    expect(isFindShortcut({ type: "keydown", ctrlKey: true, metaKey: false, key: "c" })).toBe(false);
  });
});
